libraries/standard/json/cbmc/proofs/JSON_Validate/glue.c

```diff
@@ -0,0 +1,45 @@
+#include "glue.h"
+
+bool_ skipAnyLiteral( const char * buf,
+                      size_t * start,
+                      size_t max )
+{
+    /* min argument is 4 for the shortest literal, e.g., true or null. */
+    return skipGeneric( buf, start, max, 4 );
+}
+
+JSONStatus_t skipCollection( const char * buf,
+                             size_t * start,
+                             size_t max )
+{
+    JSONStatus_t ret;
+
+    __CPROVER_assume( ( ret == JSONPartial ) || ( ret == JSONIllegalDocument ) || ( ret == JSONMaxDepthExceeded ) );
+
+    /* min argument is 2 for an empty collection, i.e., {} or []. */
+    return ( skipGeneric( buf, start, max, 2 ) == true ) ? JSONSuccess : ret;
+}
+
+bool_ skipNumber( const char * buf,
+                  size_t * start,
+                  size_t max )
+{
+    /* min argument is 1 for a single digit, e.g., 0. */
+    return skipGeneric( buf, start, max, 1 );
```

Comment:
It's expressing the shortest successful match the original function could return.  The skipGeneric() only uses the value as a lower bound.  A 0 or negative here would break parity with the original function.